src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,202 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(null, value, null);
+        if (size == 0) {
+            head = newNode;
+            tail = newNode;
+        } else {
+            tail.next = newNode;
+            newNode.prev = tail;
+            tail = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        isIndexValidToAddNode(index);
+        Node<T> oldNode = findByIndex(index);
+        Node<T> newNode = new Node<>(null, value, null);
+
+        if (index == 0 && size != 0) {
+            head.prev = newNode;
+            newNode.next = head;
+            head = newNode;
+            size++;
+        } else if (index == 0) {
+            add(value);
+        } else if (index == size) {
+            tail.next = newNode;
+            newNode.prev = tail;
+            tail = newNode;
+            size++;
+        } else if (oldNode != null) {
+            oldNode.prev.next = newNode;
+            newNode.prev = oldNode.prev;
+
+            newNode.next = oldNode;
+            oldNode.prev = newNode;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T item : list) {
+            add(item);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        isIndexValid(index);
+        return findByIndex(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        isIndexValid(index);
+        Node<T> node = findByIndex(index);
+        T oldValue = node.value;
+        node.value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        isIndexValid(index);
+        Node<T> node = findByIndex(index);
+        unlink(node);
+        size--;
+        return node.value;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> node = findByValue(object);
+        if (node != null) {
+            unlink(node);
+            size--;
+            return true;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    /**
+     * Removes all connections of a certain node
```

Comment:
is this comments are really reqired, and if yes why other methods not documented